primary/src/block_synchronizer/mod.rs

```diff
@@ -40,7 +43,17 @@ pub mod responses;
 /// proceed to next state.
 const CERTIFICATE_RESPONSES_RATIO_THRESHOLD: f32 = 0.5;
 
-type ResultSender<T> = Sender<BlockSynchronizeResult<Certificate<T>>>;
+#[derive(Debug, Clone)]
+pub struct BlockHeader<PublicKey: VerifyingKey> {
```

Comment:
I've addressed the rest of comments so If you are ok to keep the `BlockHeader` as is (given the above) do you mind approving as this will allow me to progress with follow up work. Otherwise let's discuss further about the `BlockHeader`.

Reply:
I'm ok with temporizing on this. Please open an issue: essentially, we want `BlockHeader` to take different branches depending on its flag, and ideally not just in tests.

There are ultimately several prettier ways of doing this, but we'll discuss the stylistic details once we have a clearer picture of the control flow.